src/dxtbx/model/experiment_list.py

```diff
@@ -671,7 +672,13 @@ def from_filenames(
         format_groups = collections.defaultdict(list)
         if format_kwargs is None:
             format_kwargs = {}
-        for filename in to_process:
+
+        if os.isatty and len(filenames) > 1:
```

Comment:
This is moving from accepting generator to requiring a list - should there be a check for this or type hints?

Reply:
Just cast to a list, since this is the least harmful change